src/Microsoft.OData.Core/JsonLight/ODataJsonLightSerializer.cs

```diff
@@ -69,13 +71,13 @@ internal ODataJsonLightSerializer(ODataJsonLightOutputContext jsonLightOutputCon
             // the two separate instances when asynchronous API implementation is in progress
             this.odataAnnotationWriter = new SimpleLazy<JsonLightODataAnnotationWriter>(
                 () => new JsonLightODataAnnotationWriter(
-                    jsonLightOutputContext.JsonWriter,
-                    this.JsonLightOutputContext.OmitODataPrefix,
+                    this.jsonLightOutputContext.JsonWriter,
+                    this.jsonLightOutputContext.OmitODataPrefix,

```

Comment:
I assume these changes are "just" for improved consistency, or are they also improving perf?

Reply:
@habbes The changes here are not cosmetic. Passing the local `jsonLightOutputContext` was causing state from the enclosing context to be captured.

**Before:**
![image](https://user-images.githubusercontent.com/28291332/190969526-21ced216-502b-4533-8265-13e5b40e6557.png)

**After:**
![image](https://user-images.githubusercontent.com/28291332/190969697-cdeae7e6-deab-4224-a7e4-85ca3ebb75bd.png)